### ISSUE TITLE
Table column aria-label implementation creates a poor screen reader experience

### ISSUE THREAD

**Issue by mellis481** [ant-design/ant-design#39410]
### Reproduction link

[https://ant.design/components/table#components-table-demo-head](https://ant.design/components/table#components-table-demo-head)

### Steps to reproduce

1. Turn on screen reader.
2. Go to example URL.
3. Tab to Age column.
4. See the horrible column label announced.

### What is expected?

Here is a great example of how to implement an accessible table with sortable columns: https://www.w3.org/WAI/ARIA/apg/example-index/table/sortable-table. When you tab to the sortable Last Name column, simply because of the use of semantic HTML and correct aria attributes, the screen reader announces "column 2  sorted ascending  Last Name  button".

### What is actually happening?

When you tab into the Age column in the Ant example the screen reader announces "this column's title is Age, this column is sortable column header". This label is completely unnecessary and created an unexpected and confusing experience for screen reader users.

| Environment | Info |
| --- | --- |
| antd | 5.0.0 |
| React | react |
| System | Windows 10 |
| Browser | All |

---
A column does not need any aria-label if semantic HTML and correct aria attributes are used. This issue had been previously introduced in the Table component, but was made exponentially worse with #38389. Here is the specific change: https://github.com/ant-design/ant-design/pull/38389/files#diff-33a34c8ef463c5e1cad14a8046c87aa47265d4927644753c226a65211511dc41

**Comment by dgreene1** [comment]
@afc163 since you merged the PR that caused the accessibility regression, can you please introduce one or more of the following preventative measures?

- adopt an automated approach to accessibility scanning. Even though AXE plugins only catch a small percent of accessibility issues, it’s still a good “first line of defense.”
- add a checkbox to the PR process that says “I have tested this in a screen reader”
- have a member of the AntD team test it in a screen reader before merging code?

Our team (composed of @mellis481 and others) only has so much budget to fix accessibility bugs in the open source software that we use, so we need AntD to do better at maintaining the current level of accessibility. We have completed ~24 accessibility PRs and we can not allow for those to regress. So please share how this type of bug will be prevented in the future.

**Comment by afc163** [comment]
> So please share how this type of bug will be prevented in the future.

Because lack of knowledge and practice about a11y, sometimes we do it not in a perfect way. We are quite willing to accept PRs and code review.

**Comment by mellis481** [comment]
> Because lack of knowledge and practice about a11y, sometimes we do it not in a perfect way. We are quite willing to accept PRs and code review.

@afc163 I appreciate your honestly.

Inaccessible solutions prevent countless users from successfully using web apps and open up the web app owners to multi-million dollar lawsuits. This is why (as I assume you've realized) focus on accessibility is becoming more and more significant/common and a requirement for many web app owners.  Arguably the most important part of a web app to be accessible are the building blocks (widgets).  Have you or any of the other maintainers considering taking accessibility training to help ensure components you produce are accessible?  It would benefit ant-design, all the rc-* libraries you maintain, and anything else you're ever involved with.  Accessibility development is a very marketable skill.

**Comment by mellis481** [comment]
@kiner-tang Are you willing to fix the issue I described above that was introduced in your PR?

**Comment by kiner-tang** [comment]
> @kiner-tang Are you willing to fix the issue I described above that was introduced in your PR?

👌🏻
